demo-app/src/main/java/org/openmined/syft/demo/ui/MainActivity.kt

```diff
@@ -1,53 +1,75 @@
 package org.openmined.syft.demo.ui
 
+import android.opengl.Visibility
 import android.os.Bundle
+import android.widget.ProgressBar
 import androidx.appcompat.app.AppCompatActivity
 import androidx.databinding.DataBindingUtil
+import androidx.lifecycle.Observer
 import io.reactivex.Scheduler
 import io.reactivex.android.schedulers.AndroidSchedulers
 import io.reactivex.schedulers.Schedulers
+import kotlinx.android.synthetic.main.activity_main.progressBar
 import kotlinx.android.synthetic.main.activity_main.toolbar
 import org.openmined.syft.demo.R
 import org.openmined.syft.demo.databinding.ActivityMainBinding
 import org.openmined.syft.demo.datasource.LocalMNISTDataDataSource
 import org.openmined.syft.demo.domain.MNISTDataRepository
-import org.openmined.syft.demo.domain.MNISTTrainer
+import org.openmined.syft.domain.LocalConfiguration
 import org.openmined.syft.threading.ProcessSchedulers
 
+private const val TAG = "MainActivity"
+
 @ExperimentalUnsignedTypes
+@ExperimentalStdlibApi
 class MainActivity : AppCompatActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         val binding: ActivityMainBinding =
                 DataBindingUtil.setContentView(this, R.layout.activity_main)
         setSupportActionBar(toolbar)
-        binding.viewModel = initiateViewModel("192.168.1.13:5000")
+        binding.lifecycleOwner = this
+        binding.viewModel = initiateViewModel("10.0.2.2:5000")
```

Comment:
Also can you add a line chart from https://github.com/PhilJay/MPAndroidChart? It is pretty easy and just 3-4 lines. Instead of showing the logger we can have dynamically populated graph